### DISCUSSION TITLE
Estrutura de arquivos para testes unitários

### DISCUSSION THREAD

**Discussion by thiamsantos** [frontendbr/forum#362] [Perguntas e Respostas]
Onde vocês colocam normalmente os arquivos de testes?
Numa pasta separada com todos os testes, junto com o código fonte talvez adicionando um sufixo no arquivo como `.spec`, ou de outra maneira?

Ultimamente tenho me deparado com esse dilema. Quais são as suas escolhas e por quê?

**Comment by robsonbittencourt** [comment]
Eu crio uma pasta chamada test. Já os nomes dos arquivos mantenho o mesmo do arquivo sob teste.

**Comment by wmartins** [comment]
Faço o mesmo que o @robsonbittencourt.

**Comment by felipefialho** [comment]
O mesmo que o @robsonbittencourt também. 👊

**Comment by lex0c** [comment]
Eu costumo deixar os testes em uma pasta na raiz chamada 'tests', dependendo da quantidade de classes para testar divido em subdiretórios...

exemplo:

tests:
- bd:
  - ConectionTest
  - CRUDTest
  - ...
-  requests:
  - InputTest
  - ...
- etc:
  - ...

Sempre em camelcase e colocando 'Test' no final 'NomedaClasseTest.php'

**Comment by thiamsantos** [comment]
@robsonbittencourt @wmartins @LFeh eu costumava fazer igual a vocês.

Mas eu li em algum lugar o medium um artigo sugerindo essa abordagem:

```
└── js
    └── module-one.js
    └── module-one.spec.js
    └── module-two.js
    └── module-two.spec.js
```

E até agora estou bem feliz, porque mantém os testes junto com o código.

**Comment by matheusml** [comment]
@thiamsantos talvez você esteja falando desse post: [Organizando uma aplicação com React
](https://medium.com/@matheusml/organizando-uma-aplica%C3%A7%C3%A3o-com-react-5b8ea9075596#.301wpiqdz)

Eu também utilizava a mesma abordagem citada aqui. Uma pasta `test` separada.
Mas na minha visão o ideal é termos o teste do componente juntamente com o próprio componente:

```
components/
  home/
    Home.js
    Home.spec.js
    styles.css
```

Dessa forma, quando alteramos um componente, já temos a certeza que o teste tá ali do lado pra ser mudado também. E se deletarmos a _feature_ inteira, o teste vai junto.
